tests/js/searchtools.js

```diff
@@ -27,6 +27,36 @@ describe('Basic html theme search', function() {
       expect(Search.performTermsSearch(searchterms, excluded, terms, titleterms)).toEqual(hits);
     });
 
+    it('should be able to search for multiple terms', function() {
+      index = {
+        alltitles: {
+          'Main Page': [[0, 'main-page']],
+        },
+        docnames:["index"],
+        filenames:["index.rst"],
+        indexentries:{},
+        objects:{},
+        objtypes: {},
+        objnames: {},
+        terms:{main:0, page:0},
+        titles:["Main Page"],
+        titleterms:{ main:0, page:0 }
+      }
+      Search.setIndex(index);
+      searchterms = ['main', 'page'];
+      excluded = [];
+      terms = index.terms;
+      titleterms = index.titleterms;
+      hits = [[
+        'index',
+        'Main Page',
+        '',
+        null,
+        7,
```

Comment:
Should be changed to 15 once #11958 lands

```suggestion
        15,
```

Reply:
Should we wait instead for that PR to land then? If not, add a TODO note.

Reply:
Yeah, my preference would be to land #11958, then rebase this one on top of that (with the above change applied). I'll update the PR description with a note.